src/frequenz/sdk/microgrid/component_data.py

```diff
@@ -17,229 +19,218 @@
 from .component_states import EVChargerCableState
 
 
-class _BaseComponentData(ABC):
+@dataclass(frozen=True)
+class ComponentData(ABC):
     """A private base class for strongly typed component data classes."""
 
-    _raw: microgrid_pb.ComponentData
+    component_id: int
+    timestamp: datetime
 
-    @property
-    def component_id(self) -> int:
-        """Return the component id of the message.
+    @staticmethod
+    def from_proto(raw: microgrid_pb.ComponentData) -> ComponentData:
+        """Create ComponentData from a protobuf message.
 
-        Returns:
-            Component id of the message.
+        Args:
+            raw: protobuf message
         """
-        return self._raw.id
-
-    @property
-    def timestamp(self) -> datetime:
-        """Return the timestamp of the message.
 
-        Returns:
-            Timestamp of the message.
-        """
-        return self._raw.ts.ToDatetime(tzinfo=pytz.UTC)
 
+@dataclass(frozen=True)
+class MeterData(ComponentData):
+    """A wrapper class for holding meter data.
 
-class MeterData(_BaseComponentData):
-    """A wrapper class for holding meter data."""
+    component_id: the component ID
```

Comment:
The documentation is is pretty obvious, but maybe make it `component_id: the ID identifying this component in the microgrid`